### PR TITLE
daemon: move check for CPU-realtime daemon options

### PR DESCRIPTION
**Pull request by thaJeztah**
follow-up to https://github.com/moby/moby/pull/43130

Perform the validation when the daemon starts instead of performing these
validations for each individual container, so that we can fail early.

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: daemon/oci_linux.go [hunk 1]
```diff
@@ -818,16 +818,6 @@ func WithCgroups(daemon *Daemon, c *container.Container) coci.SpecOpts {
...
-		if cdcgroups.Mode() == cdcgroups.Unified {
-			return errors.New("daemon-scoped cpu-rt-period and cpu-rt-runtime are not implemented for cgroup v2")
-		}
-
-		// FIXME this is very expensive way to check if cpu rt is supported
-		sysInfo := daemon.RawSysInfo()
-		if !sysInfo.CPURealtime {
-			return errors.New("daemon-scoped cpu-rt-period and cpu-rt-runtime are not supported by the kernel")
-		}
-
```
Comment (thaJeztah) [inline daemon/oci_linux.go:830]: If we have merged https://github.com/moby/moby/pull/43130, we _could_ consider to keep the check here as well, but I don't think we would need it if we make the daemon fail to start with these options set

**Comment by thaJeztah** [comment]
Hmmm... looks like this is failing on Windows; possibly missing build-tags somewhere;

```

[2022-01-07T23:31:49.530Z] Building: bundles/cross/windows/amd64-daemon/dockerd-dev.exe
[2022-01-07T23:31:49.530Z] GOOS="windows" GOARCH="amd64" GOARM=""
[2022-01-07T23:31:57.647Z] # github.com/docker/docker/vendor/github.com/containerd/cgroups
[2022-01-07T23:31:57.647Z] vendor/github.com/containerd/cgroups/memory.go:463:14: undefined: unix.Eventfd
[2022-01-07T23:31:57.647Z] vendor/github.com/containerd/cgroups/memory.go:463:30: undefined: unix.EFD_CLOEXEC
[2022-01-07T23:31:57.647Z] vendor/github.com/containerd/cgroups/memory.go:467:15: cannot assign error to err in multiple assignment
[2022-01-07T23:31:57.647Z] vendor/github.com/containerd/cgroups/memory.go:469:3: undefined: unix.Close
[2022-01-07T23:31:57.647Z] vendor/github.com/containerd/cgroups/memory.go:476:3: undefined: unix.Close
[2022-01-07T23:31:57.647Z] vendor/github.com/containerd/cgroups/utils.go:64:10: undefined: unix.Statfs_t
[2022-01-07T23:31:57.647Z] vendor/github.com/containerd/cgroups/utils.go:65:13: undefined: unix.Statfs
[2022-01-07T23:31:57.647Z] vendor/github.com/containerd/cgroups/utils.go:70:8: undefined: unix.CGROUP2_SUPER_MAGIC
[2022-01-07T23:31:57.647Z] vendor/github.com/containerd/cgroups/utils.go:74:14: undefined: unix.Statfs
[2022-01-07T23:31:57.647Z] vendor/github.com/containerd/cgroups/utils.go:77:18: undefined: unix.CGROUP2_SUPER_MAGIC
[2022-01-07T23:31:57.647Z] vendor/github.com/containerd/cgroups/utils.go:77:18: too many errors
```

**Comment by thaJeztah** [comment]
@tianon @AkihiroSuda ptal

File: cmd/dockerd/daemon_linux.go [hunk 2]
```diff
@@ -16,3 +22,20 @@ func notifyReady() {
...
+func validateCPURealtimeOptions(config *config.Config) error {
+	if config.CPURealtimePeriod == 0 && config.CPURealtimeRuntime == 0 {
+		return nil
+	}
+
+	if cdcgroups.Mode() == cdcgroups.Unified {
+		return errors.New("daemon-scoped cpu-rt-period and cpu-rt-runtime are not implemented for cgroup v2")
+	}
+
+	// FIXME this is very expensive way to check if cpu rt is supported
```
Comment (tianon) [inline cmd/dockerd/daemon_linux.go:35]: Is it expensive enough that we should skip it for now or just that we should think about maybe finding an alternative "eventually" ? (Given that this is a comment and the check is here, I'm assuming the latter? :see_no_evil:)

**Review by tianon — APPROVED** [review]

File: cmd/dockerd/daemon_linux.go (hunk 2 above)
Comment (thaJeztah) [reply cmd/dockerd/daemon_linux.go]: Removed the TODO / comment 👍

**Review by tianon — APPROVED** [review]

**Comment by thaJeztah** [comment]
@AkihiroSuda PTAL

**Review by AkihiroSuda — APPROVED** [review]

**Comment by thaJeztah** [comment]
Thanks @AkihiroSuda ! If you find some time at some point, perhaps you could have a look at https://github.com/moby/moby/issues/43213 (you probably know more about that from the top of your head than I do 😅)

**Pull request merged**